.claude/hooks/gemini-context-injector.sh

```
@@ -0,0 +1,129 @@
+# Function to log injection events
+log_injection_event() {
+    local event_type="$1"
+    local details="$2"
+    local timestamp=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+    echo "{\"timestamp\": \"$timestamp\", \"event\": \"$event_type\", \"details\": \"$details\"}" >> "$LOG_FILE"
+}
+
+# Main logic
+main() {
+    # Extract tool information from stdin
+    local tool_name=$(echo "$INPUT_JSON" | jq -r '.tool_name // ""')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)